### PR TITLE
fix(iOS): modal not presenting when deep in stack

### PR DESCRIPTION
**Pull request by maksg**
## Description

When 3 screens with a ‘modal’ presentation mode are stacked, navigating to a screen outside of the current stack is not functioning correctly.

## Changes

Removed dismissing logic for `lastModalVc`.

### Before
https://github.com/user-attachments/assets/d4991253-52f1-4f35-8698-242d98b218d1

### After
https://github.com/user-attachments/assets/571ebd45-7319-4225-ab5b-4f68027f4f09

## Test code and steps to reproduce

`TestModalNavigation` was added to tests.

## Checklist

- [x] Included code example that can be used to test this change
- [ ] Ensured that CI passes

### CODE REVIEW

File: ios/RNSScreenStack.mm [hunk 1]
```diff
@@ -513,13 +513,6 @@ - (void)setModalViewControllers:(NSArray<UIViewController *> *)controllers
       [changeRootController dismissViewControllerAnimated:shouldAnimate completion:finish];
       return;
     }
-
-    UIViewController *lastModalVc = [self lastFromPresentedViewControllerChainStartingFrom:firstModalToBeDismissed];
-
-    if (lastModalVc != firstModalToBeDismissed) {
-      [lastModalVc dismissViewControllerAnimated:shouldAnimate completion:finish];
-      return;
-    }
```
Comment (tboba) [inline ios/RNSScreenStack.mm:522]: Have you checked if this doesn't break anything else? According to blame, this check has been here added in purpose. See: https://github.com/software-mansion/react-native-screens/commit/471127e26aa23eed4d14388a9646bf83de5e8699
Could you check if `Test1829.tsx` works properly with changes from this PR?

Comment (maksg) [reply ios/RNSScreenStack.mm:522]: Yes, we had a call with @kkafar trying to figure out the proper fix prior to making this PR and we checked a few scenarios including `Test1829`.
Unless there is some hidden scenario which breaks by this, there shouldn't be any issues with the removal of those lines.

**Review by kkafar — COMMENTED** [review]
Hey, I confirm that we tested this code and it seems that some other changes in code made the to-be-removed lines obsolete or even "misbehaving" (they introduce a bug now, instead of improving the situation). 
We've also tested common cases and navigation scenarios & it seems we're good on this front. 

Let's solve the merge conflicts and we can proceed.

**Comment by kkafar** [comment]
Also for some reason iOS e2e CI is failing - I've restarted it. Let's make sure it passes.

**Comment by maksg** [comment]
@kkafar iOS tests seem to be failing since last week so I'm not sure what's exactly the cause of that.

Anyway, I've resolved conflicts.

**Review by kkafar — APPROVED** [review]
Let's go. Thanks for these changes.

**Pull request merged**